src/librustc/mir/interpret/mod.rs

```diff
@@ -554,3 +556,70 @@ pub fn truncate(value: u128, size: Size) -> u128 {
     // Truncate (shift left to drop out leftover values, shift right to fill with zeroes).
     (value << shift) >> shift
 }
+
+
+impl<'tcx> TyCtxt<'tcx> {
+
+    /// Evaluates a constant without providing any substitutions. This is useful to evaluate consts
+    /// that can't take any generic arguments like statics, const items or enum discriminants.
+    pub fn const_eval_mono(self, def_id: DefId) -> ConstEvalResult<'tcx> {
+        let substs = InternalSubsts::identity_for_item(self, def_id);
+        let instance = ty::Instance::new(def_id, substs);
+        let cid = GlobalId {
+            instance,
+            promoted: None,
+        };
+        let param_env = if self.is_static(def_id) {
```

Comment:
`tcx.param_env(def_id)` should still work for statics, it's better not to special-case things.